src/main/java/hudson/plugins/sauce_ondemand/SauceEnvironmentUtil.java

```diff
@@ -209,4 +212,10 @@ public static String getBuildName(AbstractBuild<?, ?> build) {
         }
         return builder.toString();
     }
+
+    public static String generateTunnelIdentifier(final AbstractBuild build) {
```

Comment:
making this function dependent on Build class makes it much harder to test because you need to generate projects and use mocks even though the only thing you care about is String really.

deciding what string to get from a Build is one responsibility (bound to Jenkins object model).
deciding how to process that string is another responsibility, which is not related to Jenkins object model at all.

also, I think you only call this method once...

also, as you can see below, you probably don't need any of that. if you use UUID.randomUUID(), you can delete 90% of the code in this pull request and it will still work fine with unique IDs.